base/workflow/inst/batch_runs.R

```diff
@@ -0,0 +1,431 @@
+    collect()
+  outdir_pre <- paste(
+    model.new[["model_name"]],
+    format(as.Date(start_date), "%Y-%m"),
+    format(as.Date(end_date), "%Y-%m"),
+    met, site_id, "test_runs",
+    sep = "_"
+  )
+  outdir <- file.path(output_folder, outdir_pre)
+  dir.create(outdir, showWarnings = FALSE, recursive = TRUE)
+  # Convert to absolute path so I don't end up with unnecessary nested
+  # directories
+  outdir <- normalizePath(outdir)
+  settings$outdir <- outdir
+
+  #Database BETY
+  settings$database <- list(
+    bety = list(user = db_bety_username,
+                password = db_bety_password,
+                host = db_bety_hostname,
+                dbname = "bety",
+                driver = db_bety_driver,
+                write = FALSE),
+    dbfiles = dbfiles_folder
+  )
+
+  #PFT
+  if (is.null(pft_name)){
+    # Select the first PFT in the model list.
+    pft <- tbl(bety, "pfts") %>%
+      filter(modeltype_id == !!model.new$modeltype_id) %>%
+      collect()
+    pft_name <- pft$name[[1]]
+  }
+  settings$pfts <- list(
+    pft = list(name = pft_name,
+               constants = list(num = 1))
+  )
+
+  #Meta Analysis
+  settings$meta.analysis <- list(iter = 3000, random.effects = FALSE)
+
+  #Ensemble
+  settings$ensemble <- list(
+    size = ensemble_size,
+    variable = ensemble_variable,
+    samplingspace = list(met = list(method = "sampling"),
+                         parameters = list(method = "uniform"))
+  )
+
+  #Sensitivity 
+  if (sensitivity) {
+    settings$sensitivity.analysis <- list(
+      quantiles = list(sigma1 = -2, sigma2 = -1, sigma3 = 1, sigma4 = 2)
+    )
+  }
+
+  #Model
+  settings$model$id <- model.new[["id"]]
+
+  #Workflow
+  settings$workflow$id
+  settings$workflow$id <- paste0("Test_run_","_",model.new$model_name)
+  settings$run <- list(
+    site = list(id = site_id, met.start = start_date, met.end = end_date),
+    inputs = list(met = list(source = met, output = model.new[["model_name"]],
+                             username = "pecan")),
+    start.date = start_date, end.date = end_date
+  )
+  settings$host$name <- "localhost"
+
+  #create file and Run
+  saveXML(listToXml(settings, "pecan"), file = file.path(outdir, "pecan.xml"))
+  file.copy(file.path(pecan_path, "web", "workflow.R"), outdir)
+  cwd <- getwd()
+  setwd(outdir)
+  on.exit(setwd(cwd), add = TRUE)
+
+  sys_out <- system("Rscript workflow.R 2>&1 | tee workflow.Rout")
+
+  list(
+    sys = sys_out,
+    outdir = outdir
+  )
+}
+
+library(tidyverse)
+library(PEcAn.DB)
+library(PEcAn.utils)
+library(XML)
+library(PEcAn.settings)
+
+argv <- commandArgs(trailingOnly = TRUE)
+
+##Create Run Args
+met_name <- c("CRUNCEP", "AmerifluxLBL")
+startdate <- "2004-01-01"
+enddate <- "2004-12-31"
+out.var <- "NPP"
+ensemble <- FALSE
+ens_size <- 1 # Run ensemble analysis for some models?
+sensitivity <- FALSE
+user_id <- 99000000002   # TODO: Un-hard-code this
+dbfiles_folder <- normalizePath("~/output/dbfiles") # TODO: Un-hard-code this
```

Comment:
Generally it's better to have library loads and run args at the top of a script, not this far down. The function def'n above this really would be better in it's own file (which is consistent with previous comment about resolving duplication)